app/models/list_account.rb

```diff
@@ -25,17 +25,16 @@ class ListAccount < ApplicationRecord
   private
 
   def set_follow
-    self.follow = Follow.find_by!(account_id: list.account_id, target_account_id: account.id)
-  rescue ActiveRecord::RecordNotFound
```

Comment:
I think the CI failure here - https://github.com/mastodon/mastodon/actions/runs/12042045495/job/33575078943?pr=33082 - is that the controller was implicitly relying on this raise to convert into a 404...

Reply:
I'm not sure if claiming 404 in the API is the desired outcome. The `validate_relationship` method was supposed to give more specific error messages but the code was never reached because of the exception...

Reply:
Yeah ... sort of related - https://github.com/mastodon/mastodon/pull/31896 - on that PR I had troubled (because of the combination of the raise and the factory setup) adding any non-convoluted coverage for this latter method. I was able to do the follow setting, and meant to do the validation part as followup.

It would definitely be easier to accomplish that if the raise comes out.

LMK if pulling out the raise and adding validation coverage (but not yet changing the API or this validation message string) in advanced of this would be useful...?